### DISCUSSION TITLE
Using LiteDb on client Side

### DISCUSSION THREAD

**Discussion by Arslan007** [Azure/azure-mobile-apps#507] [Q&A]
We are using LiteDB for our next MAUI project and wants to implement offline sync in that app.
I know Azure Mobile apps currently supports SQLite and doesn't support LiteDB due to lack of .Net 6 support in LiteDB.
Is there any guidance to start this implementation or there can be some bottlenecks/challenges that can be hurdle for using LiteDB in client side for Offline Sync.

Thanks!

**Comment by adrianhall** [comment]
I'd take the SQLite version of the offline database and work through the SQLite specific areas.

There are three specific things you need to look at:

1. How to represent date/times with ms accuracy and comparison support.  SQLite doesn't support it, so we had to invent something (basically, we store the date/time as a long and convert to/from before comparing).
2. How to establish the database
3. How to do searches.  We have an OData to SQL transform built into the library.  This needs to be coded for the LiteDB specification.

I'm currently doing a Blazor sample, but I think this would be a worthwhile implementation and I'm happy to accept a PR.
